client/post-editor/editor-sharing/container.jsx

```diff
@@ -0,0 +1,81 @@
+/**
+ * External dependencies
+ */
+import React, { Component, PropTypes } from 'react';
+import { connect } from 'react-redux';
+
+/**
+ * Internal dependencies
+ */
+import PostEditStore from 'lib/posts/post-edit-store';
+import { fetchConnections } from 'lib/sharing/publicize/actions';
+import { getConnectionsBySiteId, hasFetchedConnections } from 'lib/sharing/publicize/selectors';
+import EditorSharingAccordion from './accordion';
+
+class EditorSharingContainer extends Component {
+	constructor( props ) {
+		const { site, state } = props;
+
+		super( props );
+
+		// Trigger connection fetch
+		if ( ! hasFetchedConnections( state, site.ID ) ) {
+			this.fetchConnections();
+		}
+
+		// Set state
+		this.state = this.getState();
+
+		// Bind legacy store update handler
+		this.boundUpdateState = this.updateState.bind( this );
+		PostEditStore.on( 'change', this.boundUpdateState );
+	}
+
+	componentWillUnmount() {
+		PostEditStore.off( 'change', this.boundUpdateState );
+	}
+
+	updateState() {
+		this.setState( this.getState() );
+	}
+
+	getState() {
+		return {
+			post: PostEditStore.get(),
+			isNew: PostEditStore.isNew()
+		};
+	}
+
+	fetchConnections() {
+		const { site, dispatch } = this.props;
+		dispatch( fetchConnections( site.ID ) );
+	}
+
+	render() {
+		const { site, state } = this.props;
+		const { post, isNew } = this.state;
+		const connections = getConnectionsBySiteId( state, site.ID );
+
+		return (
+			<EditorSharingAccordion
+				site={ site }
+				post={ post }
+				isNew={ isNew }
+				connections={ connections }
+				fetchConnections={ this.fetchConnections.bind( this ) }
+				className="editor-drawer__accordion" />
+		);
+	}
+}
+
+EditorSharingContainer.propTypes = {
+	site: PropTypes.object.isRequired,
+	state: PropTypes.object.isRequired,
+	dispatch: PropTypes.func.isRequired
+};
+
+export default connect(
+	( state ) => {
+		return { state };
+	}
+)( EditorSharingContainer );
```

Comment:
>Is the state mapping function always going to be the same?

This raises a good point, as it should depend on the data needs of the component. The [`react-redux` docs examples](https://github.com/rackt/react-redux/blob/master/docs/api.md#connectmapstatetoprops-mapdispatchtoprops-mergeprops-options) may help to shed some light here. In this case, rather than injecting the entire state tree, I should probably leverage the existing selectors, specifying `connections` and `hasFetchedConnections` as the props passed by the `connect` mapping function.

>I'm also finding it pretty confusing that state is used to refer to the state of the Redux store - this.props.state is a strange thing to see in the code.

"State" is the terminology [used by Redux](http://redux.js.org/docs/api/Store.html#getState), though I do agree that it becomes a bit confusing in this example where I'm injecting a `state` prop into the the component. In consideration of the above, this will likely not be as big a concern if I choose to not pass the entire state tree in to the component, and instead pass only the subset of data needed by the component.